packages/xdg_icons/lib/src/icon.dart

```diff
@@ -131,15 +133,31 @@ class XdgIconState extends State<XdgIcon> {
     final file = File(_icon?.fileName ?? '');
     final size = _resolveSize();
     if (file.existsSync()) {
-      final builder = _icon!.isScalable ? SvgPicture.file : Image.file;
-      return builder(
+      if (_icon!.isScalable) {
+        return ScalableImageWidget.fromSISource(
+          si: ScalableImageSource.fromSvgFile(
+            file,
+            file.readAsString,
+          ),
+        );
+      }
+
+      return Image.file(
         file,
         width: size.toDouble(),
         height: size.toDouble(),
       );
     } else if (_icon?.data != null) {
-      final builder = _icon!.isScalable ? SvgPicture.memory : Image.memory;
-      return builder(
+      if (_icon!.isScalable) {
+        return ScalableImageWidget.fromSISource(
+          si: ScalableImageSource.fromSvgFile(
+            File(''),
```

Comment:
This needs to pass something that works for equality comparison so that `jovial_svg` can correctly determine whether it needs to reload the source. I was going to suggest simply passing `file`, but it looks like equality comparison of non-existent resource paths doesn't work (`File('/org/gtk/foo/bar.svg')` is not equal to `File('/org/gtk/foo/bar.svg')`) so perhaps pass the path then?

```suggestion
            file.path,
```

Reply:
Ah, I misunderstood. `file.path` should work then, since it's just a simple string.